rust/src/fvm/types.rs

```diff
@@ -14,10 +14,12 @@ pub enum FvmRegisteredVersion {
 #[derive_ReprC]
 #[ReprC::opaque]
 #[derive(Default)]
-pub struct FvmMachine {
+pub struct InnerFvmMachine {
     pub(crate) machine: Option<Mutex<CgoExecutor>>,
 }
 
+pub type FvmMachine = Option<repr_c::Box<InnerFvmMachine>>;
```

Comment:
This is a very confusing type.

Reply:
not a great type, but necessary, I'll add a comment